classes/Product.php

```diff
@@ -6072,6 +6072,10 @@ public function getCoverWs()
     {
         $result = $this->getCover($this->id);
 
+        if (!is_array($result)) {
+            return null;
```

Comment:
`null` does not match the `@return` statement in the PHP doc. We expect integers to be returned.

Reply:
I'd prefer to read 0 than null here, not really fan of multiple return types when we can avoid it: wdyt?

Reply:
I would agree with you @mickaelandrieu but since it is an ID and some migration tools (or teams) use zeros or negatives for the IDs introduced by them I thought ```null``` fits better. Also I'm against to fake data, unless the data represents a sort value, or position, etc...

Reply:
In a XML I guess the value won't be rendered the same way.

0 => `string(13) "<![CDATA[0]]>"`
null => `string(12) "<![CDATA[]]>"`

https://github.com/PrestaShop/PrestaShop/blob/c33d60e02e1b3270bd192aadfd85ae7a31dd562f/classes/webservice/WebserviceOutputXML.php#L138-L140

But this needs further tests to confirm.

Reply:
Note that `<![CDATA[]]>` means empty string, not null.